packages/appium/scripts/parse-yml-commands.js

```
@@ -5,55 +5,55 @@
 const path = require('path');
 const yaml = require('yaml');
 const {fs, util, logger} = require('@appium/support');
-const validate = require('validate.js');
 const Handlebars = require('handlebars');
 const _ = require('lodash');
 const {asyncify} = require('asyncbox');
 const url = require('url');
 
 const log = logger.getLogger('YamlParser');
 
-validate.validators.array = function array(value, options, key, attributes) {
-  if (attributes[key] && !validate.isArray(attributes[key])) {
-    return `must be an array`;
-  }
-};
-
-validate.validators.hasAttributes = function hasAttributes(value, options) {
-  if (!value) {
-    return;
-  }
-
-  if (!_.isArray(value)) {
-    value = [value];
-  }
+const validators = {
+  isArray: (value, options) => {
+    if (_.isUndefined(value) || _.isArray(value) || !options) {
+      return null;
+    }
+    return ` must be an array`;
+  },
+  presence: (value, options) => {
+    if (_.isUndefined(value) && options) {
+      return `must be present`;
+    }
+    return null;
+  },
+  hasAttributes: (value, options) => {
+    if (!value) {
```

Is it maybe worth checking if `options` is non-null here as well, like in `isArray`?

---

not sure what exactly you mean. Could you provide an example code?

---

I assume that we can just skip the loops if `options` is falsy:
```js
if (!value || !options) {
  return null;
}
```
For `hasPossibleAttributes` I guess a falsy `options` value should raise an error?

---

changed